openeo/extra/job_management.py

```diff
@@ -101,18 +118,22 @@ def _normalize_df(self, df: pd.DataFrame) -> pd.DataFrame:
         return df
 
     # TODO: long method with deep nesting. Refactor it to make it more readable.
-    def run_jobs(self, df: pd.DataFrame, start_job: Callable, output_file: Path):
+    def run_jobs(
+        self, df: pd.DataFrame, start_job: Callable[[], BatchJob], output_file: Path
+    ):
         """Runs jobs, specified in a dataframe, and tracks parameters.
 
         :param df:
             DataFrame that specifies the jobs, and tracks the jobs' statuses.
         :param start_job:
             A callback which will be invoked with the row of the dataframe for which a job should be started.
+            This callable should return a openeo.rest.job.BatchJob object.
```

Comment:
FYI: you can user RestructuredText references like `:py:class:`openeo.rest.job.BatchJob`  which will turn in hyperlinks in the generated API docs